lib/wallet/src/Cardano/Wallet/Write/Tx/Balance.hs

```diff
@@ -373,40 +362,15 @@ constructUTxOIndex walletUTxO =
     walletUTxOIndex = UTxOIndex.fromMap $ toInternalUTxOMap walletUTxO
     cardanoUTxO = toCardanoUTxO Cardano.shelleyBasedEra walletUTxO
 
--- | Assumes all 'UTxO' entries have addresses with key payment credentials;
--- either normal, post-Shelley credentials, or boostrap/byron credentials
--- depending on the 'k' of the supplied 'TransactionLayer'.
-allKeyPaymentCredentials
-    :: forall k. TransactionLayer k 'CredFromKeyK SealedTx -> UTxOAssumptions
-allKeyPaymentCredentials tl = UTxOAssumptions
-    { txLayer = tl
-    , inputScriptLookup = Nothing
-    , inputScriptTemplate = Nothing
-    , txWitnessTag = transactionWitnessTag tl
-    }
-
--- | Assumes all 'UTxO' entries have addresses with script payment credentials,
--- where the scripts are both derived from the 'ScriptTemplate' and can be
--- looked up using the given function.
-allScriptPaymentCredentials
-    :: (W.Address -> (CA.Script KeyHash))
-    -> ScriptTemplate
-    -> TransactionLayer SharedKey 'CredFromScriptK SealedTx
-    -> UTxOAssumptions
-allScriptPaymentCredentials scriptLookup template tl = UTxOAssumptions
-    { txLayer = tl
-    , inputScriptLookup = Just scriptLookup
-    , inputScriptTemplate = Just template
-    , txWitnessTag = transactionWitnessTag tl
-    }
-
 balanceTransaction
-    :: forall era m changeState.
-        ( MonadRandom m
-        , IsRecentEra era
-        )
+    :: forall era m changeState
+     . ( MonadRandom m
+       , IsRecentEra era
+       )
     => Tracer m BalanceTxLog
+    -> TxWitnessTag
     -> UTxOAssumptions
+    -> (TokenBundleMaxSize -> TokenBundleSizeAssessor)
```

Comment:
This introduces multiple sources of truth again. Is there a way not to in this PR?

Reply:
This is an intermediate state, I plan to remove this one in the future commits.